test/Microsoft.AspNetCore.Server.KestrelTests/ConnectionAdapterTests.cs

```diff
@@ -109,6 +108,54 @@ public class ConnectionAdapterTests
             }
         }
 
+        [Fact]
+        public async Task ThrowingOnReadDoesNotCauseUnobservedException()
+        {
+            var unobservedExceptionThrown = false;
+
+            EventHandler<UnobservedTaskExceptionEventArgs> unobservedExceptionHandler = (sender, args) =>
+            {
+                unobservedExceptionThrown = true;
+            };
+
+            try
+            {
+                TaskScheduler.UnobservedTaskException += unobservedExceptionHandler;
```

Comment:
This just looks like trouble waiting to happen. You may end up observing unobserved tasks in tests since they run in parallel.